heroic-component/src/main/java/com/spotify/heroic/tracing/RandomParentSampler.kt

```diff
@@ -0,0 +1,44 @@
+package com.spotify.heroic.tracing
+
+import io.opencensus.trace.*
+import java.util.concurrent.ThreadLocalRandom
+
+/**
+ * The random parent sampler does not use the sampling decision from a parent when the span says
+ * to sample. Instead it randomly samples spans.
+ *
+ * This will cause traces to not be complete and will likely not capture the outliers! It can
+ * still be useful to use this sampler when you have potentially tens of thousands of spans being
+ * created, as this many spans cannot be shown in any meaningful way within the current UX of
+ * many tracing solutions.
+ */
+class RandomParentSampler(private val probability: Double) : Sampler() {
+    init {
+        if (probability < 0.0 || probability > 1.0) {
+            throw IllegalArgumentException("probability must be in range [0.0, 1.0]")
+        }
+    }
+
+    override fun shouldSample(
+        parentContext: SpanContext?,
+        hasRemoteParent: Boolean?,
+        traceId: TraceId?,
+        spanId: SpanId?, name: String?,
+        parentLinks: MutableList<Span>?): Boolean {
+
+        // Keep propagating parent sampling downstream.
```

Comment:
This comment is misleading - parent sampling decisions are not propagated downstream. The default is for isSampled is `false` so the logic is really:
- parent was sampled? apply probability to child
- no parent decision? explicitly do not sample